WcaOnRails/app/models/competition.rb

```diff
@@ -331,6 +331,21 @@ def warnings_for(user)
       unless has_any_round_per_event? && schedule_includes_rounds?
         warnings[:schedule] = I18n.t('competitions.messages.schedule_must_match_rounds')
       end
+
+      if championship_warnings.any?
+        warnings = championship_warnings.merge(warnings)
+      end
+    end
+
+    warnings
+  end
+
+  def championship_warnings
+    warnings = {}
+    self.championships.each do |championship|
+      if Championship.joins(:competition).merge(Competition.visible).exists?(championship_type: championship.championship_type, Competitions: { year: self.year })
```

Comment:
We'd like to eventually drop the `day`, `month`, `year`, `end*` columns from the competitions table, so it would be nice to handle this by using the existing `start_date` and `end_date` columns instead.

Reply:
Good point.  Done in #4532